Cabal-syntax/Cabal-syntax.cabal

```diff
@@ -0,0 +1,243 @@
+
+    if impl(ghc <8.8)
+      ghc-options: -Wnoncanonical-monadfail-instances
+
+  if !impl(ghc >= 8.0)
+    -- at least one of lib:Cabal's dependency (i.e. `parsec`)
+    -- already depends on `fail` and `semigroups` transitively
+    build-depends: fail == 4.9.*, semigroups >= 0.18.3 && < 0.20
+
+  if !impl(ghc >= 7.10)
+    build-depends: void >= 0.7.3 && < 0.8
+
+  if !impl(ghc >= 7.8)
+    -- semigroups depends on tagged.
+    build-depends:  tagged >=0.8.6 && <0.9, bytestring-builder >= 0.10.8 && <0.11
+
+  exposed-modules:
+    Distribution.CabalSpecVersion
+    Distribution.Compat.Binary
+    Distribution.Compat.DList
+    Distribution.Compat.Exception
+    Distribution.Compat.Lens
+    Distribution.Compat.Newtype
+    Distribution.Compat.NonEmptySet
+    Distribution.Compat.Prelude
+    Distribution.Compat.Semigroup
+    Distribution.Compiler
+    Distribution.FieldGrammar
+    Distribution.Fields
+    Distribution.License
+    Distribution.ModuleName
+    Distribution.Package
+    Distribution.PackageDescription
+    Distribution.PackageDescription.Configuration
+    Distribution.PackageDescription.FieldGrammar
+    Distribution.PackageDescription.Parsec
+    Distribution.PackageDescription.PrettyPrint
+    Distribution.PackageDescription.Utils
+    Distribution.Pretty
+    Distribution.SPDX
+    Distribution.System
+    Distribution.Types.BuildInfo.Lens
+    Distribution.Types.Component
+    Distribution.Types.ComponentId
+    Distribution.Types.ComponentName
+    Distribution.Types.ComponentRequestedSpec
+    Distribution.Types.Condition
+    Distribution.Types.ConfVar
+    Distribution.Types.Dependency
+    Distribution.Types.ExeDependency
+    Distribution.Types.ExecutableScope
+    Distribution.Types.ForeignLib
+    Distribution.Types.GenericPackageDescription
+    Distribution.Types.GenericPackageDescription.Lens
+    Distribution.Types.IncludeRenaming
+    Distribution.Types.Library
+    Distribution.Types.LibraryName
+    Distribution.Types.LibraryVisibility
+    Distribution.Types.Mixin
+    Distribution.Types.Module
+    Distribution.Types.ModuleReexport
+    Distribution.Types.ModuleRenaming
+    Distribution.Types.MungedPackageId
+    Distribution.Types.MungedPackageName
+    Distribution.Types.PackageDescription
+    Distribution.Types.PackageDescription.Lens
+    Distribution.Types.PackageId
+    Distribution.Types.PackageId.Lens
+    Distribution.Types.PackageName
+    Distribution.Types.PackageName.Magic
+    Distribution.Types.PackageVersionConstraint
+    Distribution.Types.UnitId
+    Distribution.Types.UnqualComponentName
+    Distribution.Types.Version
+    Distribution.Types.VersionRange
+    Distribution.Types.VersionRange.Internal
+    Distribution.Utils.Generic
+    Distribution.Utils.MD5
+    Distribution.Utils.Path
+    Distribution.Utils.ShortText
+    Distribution.Utils.Structured
+    Distribution.Version
+    Language.Haskell.Extension
+
+  other-modules:
```

Comment:
Most of these modules should be `exposed-modules`. In fact, the different lists could be combiled.

I don't think any module have to be hidden (i.e. `other-module`), except maybe some `Compat` modules, though I'm not sure about these either.

Reply:
Updated